### PR TITLE
Expose build-dir config option

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This config value is used to determine the relative dir that the build artifacts are output to. It should be exposed as a config option for ease of use.

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/config.dart [hunk 1]
```diff
@@ -24,6 +24,8 @@ class ConfigCommand extends FlutterCommand {
     argParser.addOption('gradle-dir', help: 'The gradle install directory.');
     argParser.addOption('android-sdk', help: 'The Android SDK directory.');
     argParser.addOption('android-studio-dir', help: 'The Android Studio install directory.');
+    argParser.addOption('build-dir', help: 'The relative path to override a projects build directory',
+        valueHelp: '/out');
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/config.dart]: Should this be 'out/' ?

File: packages/flutter_tools/lib/src/commands/config.dart [hunk 2]
```diff
@@ -128,6 +130,9 @@ class ConfigCommand extends FlutterCommand {
     if (argResults.wasParsed('clear-ios-signing-cert'))
       _updateConfig('ios-signing-cert', '');
 
+    if (argResults.wasParsed('build-dir'))
+      _updateConfig('build-dir', argResults['build-dir']);
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/config.dart]: Is this the right place to validate that it is a relative path?

File: packages/flutter_tools/lib/src/commands/config.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/config.dart]: Fixed

File: packages/flutter_tools/lib/src/commands/config.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/config.dart]: Fixed and added test

**Review by zanderso — APPROVED** [review]

**Comment by jonahwilliams** [comment]
tool_tests-linux completed but did not phone home

**Pull request merged**